### PR TITLE
Refactor ClusterSettings page and add storybook tests

### LINKED ISSUE #951: Refactor cluster settings page code
### Feature description

Currently, the code in the ClusterSettings file is over 500 lines, which is very hard to maintain. Looking forward to pulling the settings out to separate components.

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Closes #951 

## Description

Refactor the codes in ClusterSettings.tsx and split them into different files based on the settings. No UI changes.

One tiny detailed change: Disable the submit button when the user sets the culler timeout to less than 10 mins (we only show the error message before but didn't disable the submit button)

## How Has This Been Tested?

1. Get an account with ODH admin access
2. Go to the cluster settings page of the dashboard
3. Change the settings, try to see how submit button performs and try to submit the changes to see if it works

## Test Impact

Add the storybook test for the whole page, since there is no e2e test for now, I can only test the user inputs and changes to check whether the error messages are correctly shown and check if the submit button is enabled/disabled based on the user inputs.
One thing that I addressed in this PR is I add this to all the tests.
```
configure({
  testIdAttribute: 'data-id',
});
```
We already set a lot of `data-id` in the frontend code to help with the jest test before. I think we should follow this way (compared to refactoring all the `data-id` to default `data-testid`).

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

**Review by Gkrumbach07 — COMMENTED** [review]
tests looks, changes to CI work, settings in still functioning the same after refactor. some comments on constants

File: frontend/src/pages/clusterSettings/const.ts
Comment (DaoDaoNoCode) [reply frontend/src/pages/clusterSettings/const.ts:26]: The `PVC_SIZE_HINT`, `CULLER_TIMEOUT_HINT` and `TOLERATION_FORMAT_ERROR` are used both in the page and in the storybook file to find the element. The others are not, I guess it was because they were set as the const before so I moved them to the file simply, I can make them inline if it's needed.

Comment (Gkrumbach07) [reply frontend/src/pages/clusterSettings/const.ts:26]: no need. i didnt know they were already consts

**Comment by Gkrumbach07** [comment]
changes look good

/lgtm

File: frontend/src/pages/clusterSettings/PVCSizeSettings.tsx
Comment (lucferbux) [inline frontend/src/pages/clusterSettings/PVCSizeSettings.tsx:45]: Just a small nit, if we have other regex stored as constants maybe we can add this too right?

File: frontend/src/pages/clusterSettings/CullerSettings.tsx
Comment (lucferbux) [inline frontend/src/pages/clusterSettings/CullerSettings.tsx:109]: I've asked if there's a way of controlling the style of a textinupt outside of overriding the maxWidth, but in the meantime I think it's fine

**Review by lucferbux — APPROVED** [review]
Just have one nit and a question about the width of some elements, but a part from that I think it's ready to go.

/lgtm

**Comment by andrewballantyne** [comment]
/hold

image

I don't see screenshots, I don't see UX tags. This is not ready to merge.

**Comment by DaoDaoNoCode** [comment]
@andrewballantyne Sorry, no UI changes. I already unchecked the checkbox.

**Comment by lucferbux** [comment]
@DaoDaoNoCode you might want to refactor the tests to follow the new pattern we are implementing for testing https://github.com/opendatahub-io/odh-dashboard/pull/1314 once that's done we can aim to merge it this sprint

**Comment by Gkrumbach07** [comment]
tests look good

**Comment by lucferbux** [comment]
@DaoDaoNoCode I think you need to rebase main

File: frontend/src/services/clusterSettingsService.ts
Comment (lucferbux) [inline frontend/src/services/clusterSettingsService.ts:14]: This might be out of scope for this PR, but as we are growing in functionality here, we might need to either refactor cluster settings to use the pass through API or at least refactor the API calls to the backend to make it similar to the k8s call we are performing, depending on the outcome of the conversation we might wanna track another issue as a follow up to these changes.

Comment (DaoDaoNoCode) [reply frontend/src/services/clusterSettingsService.ts:14]: hmm, I think you are right, I will mark this PR as WIP because I also want to remove it to pass-through API

**Review by lucferbux — APPROVED** [review]
/lgtm

**Comment by lucferbux** [comment]
@Gkrumbach07 can you re-review this? so we can get this done asap

**Pull request merged**
